subset/password/brute_test/src/main/java/Interrogator.java

```diff
@@ -0,0 +1,103 @@
+public class Interrogator {
+
+  String[] expected = {
+    "OFS2 login:", "Password:", "Last login:", "Login incor", "Connection closed by foreign host."
+  };
+  String[] username;
+  String[] password;
+  String usedUsername;
+  String usedPassword;
+  int usernameCount = 0;
+  int usernameTemp = 0;
+  int attemptCount = 0;
+  int passwordIndex = 0;
+  int passwordAttempts = 0;
+  String report;
+  String testName = "security.passwords";
+  Report reportHandler;
+  TelnetSocket telnetSocket;
+  boolean debug = false;
+
+  public Interrogator(
+      TelnetSocket telnetSocket,
+      String[] username,
+      String[] password,
+      String macAddress,
+      String manufacturer) {
+    this.telnetSocket = telnetSocket;
+    reportHandler = new Report();
+    this.username = username;
+    this.password = password;
+    reportHandler.addText("MAC Address : " + macAddress + "*");
+    reportHandler.addText("Manufacturer : " + manufacturer + "*");
+  }
+
+  public void receiveData(String data) {
+
+    if (debug) {
+      System.out.println(
+          java.time.LocalTime.now() + "receiveDataLen:" + data.length() + "receiveData:" + data);
+    }
+    if (data != null) {
+      Thread parseDataThread = new Thread(() -> parseData(data));

```

Comment:
Why does this need to be a thread? This use is bizarre enough that I think it merits a comment explaining why.